pupil-spa/src/app/services/azure-queue/azure-queue.service.ts

```diff
@@ -0,0 +1,59 @@
+import * as bluebird from 'bluebird';
+import { WindowRefService } from '../window-ref/window-ref.service';
+import { Injectable } from '@angular/core';
+import { TokenService } from '../token/token.service';
+
+/**
+ * Declaration of azure queue service
+ */
+@Injectable()
+export class AzureQueueService {
+  private window;
+  private queueService;
+
+  constructor(private tokenService: TokenService,
+              private windowRefService: WindowRefService) {
+    this.window = windowRefService.nativeWindow;
+  }
+
+  /**
+   * Initialise queue service for messaging
+   * @param {String} queue
+   * @param {String} tokenKey
+   * @returns {Promise}
+   */
+  private initQueueService(queue, tokenKey): void {
+    const token = this.tokenService.getToken(tokenKey);
+    if (!this.queueService) {
+      this.queueService = this.window.AzureStorage.Queue
```

Comment:
house this as an angular service/component called azureQueueService rather than referencing directly

Reply:
@GuyHarwood - this _is_ the housing... so probably alright.

Reply:
my bad 😄

Reply:
@GuyHarwood @jon-shipley the azure queue service is not depending on window ref service anymore since the implementation changed. I am resolving this conversation while awaiting to get feedback on the changes I made.